### PR TITLE
Provide optional pre-commit hook config

### LINKED ISSUE #2167: [FEA] provide automated copyright header update
**Is your feature request related to a problem? Please describe.**
Contributors to RAPIDS Accelerator for Spark  are required to update the copyright header in the files modified in pull requests. This mundane task can be easily automated.

**Describe the solution you'd like**
A git commit hook that takes care of the copyright headers before submitting a pull request: 

e.g. : 

````bash
git diff origin/branch-0.5..HEAD |  grep -F "+++ b/" |  sed "s#+++ b/##g"  | 
   xargs -L1 sed -i -E "s/Copyright *\(c\) *([0-9,-]+)*-([0-9]{4}), *NVIDIA *CORPORATION/Copyright (c) \\1-`date +%Y`, NVIDIA CORPORATION/; /`date +%Y`/! s/Copyright *\(c\) ([0-9]{4}), *NVIDIA *CORPORATION/Copyright (c) \\1-`date +%Y`, NVIDIA CORPORATION/"
````
h/t @karthikeyann 

there are also 3rd party solutions 
- https://github.com/fwolf/copyright-year-updater.sh 

**Describe alternatives you've considered**
- manual updates as before 
- semi-manual with a separate script

### PR DESCRIPTION
**Pull request by gerashegalov**
Allows to update copyrights automatically by setting the environment
variable `SPARK_RAPIDS_AUTO_COPYRIGHTER=ON`

Fixes #2167 

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -150,5 +150,15 @@ By making a contribution to this project, I certify that:
 ### Testing Your Code
 Please visit the [testing doc](tests/README.md) for details about how to run tests
 
+
+### Pre-commit hooks
+We provide a basic config `.pre-commit-config.yaml` for [pre-commit](https://pre-commit.com/) to automate some aspects of the development process. As a convenience you can enable automatic copyright year updates by following the installation instructions on the pre-commit homepage, and setting the envrionment variable:
```
Comment (jlowe) [inline CONTRIBUTING.md:155]: Please wrap to within 100 characters for improved readability in a terminal and easier line-by-line commenting during reviews.

Also I'd like to see the phrase "pre-commit homepage" made into a link to the homepage.  (Yes, it's earlier in the paragraph, but this is closer to the "follow the installation instructions" directive.)

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Comment by gerashegalov** [comment]
build

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -150,5 +150,24 @@ By making a contribution to this project, I certify that:
...
 
+
+### Pre-commit hooks
+We provide a basic config `.pre-commit-config.yaml` for [pre-commit](https://pre-commit.com/) to
+automate some aspects of the development process. As a convenience you can enable automatic 
+copyright year updates by following the installation instructions on the
+[pre-commit homepage](https://pre-commit.com/), e.g.: 
+
+```bash
+conda install -c conda-forge pre-commit
```
Comment (jlowe) [inline CONTRIBUTING.md:161]: A little surprised to see `conda` here.  There's no discussion about setting up a conda environment anywhere in this repository nor is it required to have one.  `pre-commit` installs via `pip` and a bunch of other ways already covered on their homepage, so I was not expecting to see a specific one listed here which may not apply to someone's environment and then cause confusion when they attempt it.

**Comment by gerashegalov** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Pull request merged**